File: src/main/java/pl/yalgrin/playnite/simplesync/web/RegionResource.java
package pl.yalgrin.playnite.simplesync.web;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import pl.yalgrin.playnite.simplesync.dto.RegionDTO;
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper;
import pl.yalgrin.playnite.simplesync.service.RegionService;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api/region")
@RequiredArgsConstructor
public class RegionResource {
    private final RegionService service;
    private final SingleExecutorHelper singleExecutorHelper;

    @GetMapping("/{id}")
    public Mono<RegionDTO> getRegion(@PathVariable Long id) {
        return service.findById(id);
    }

    @PostMapping("/save")
    public Mono<RegionDTO> saveRegion(@RequestBody RegionDTO dto, @RequestParam String clientId) {
        return singleExecutorHelper.runOnExecutor(service.saveObject(dto, clientId));
    }

    @PostMapping("/delete")
    public Mono<Void> deleteRegion(@RequestBody RegionDTO dto, @RequestParam String clientId) {
        return singleExecutorHelper.runOnExecutor(service.deleteObject(dto, clientId));
    }
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/web/library/RegionResource.kt
package pl.yalgrin.playnite.simplesync.web.library

import org.springframework.web.bind.annotation.*
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper
import pl.yalgrin.playnite.simplesync.library.dto.RegionDTO
import pl.yalgrin.playnite.simplesync.library.service.RegionService
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/api/region")
class RegionResource(
    private val service: RegionService,
    private val singleExecutorHelper: SingleExecutorHelper
) {

    @GetMapping("/{id}")
    fun getRegion(@PathVariable id: Long): Mono<RegionDTO> {
        return service.findById(id)
    }

    @PostMapping("/save")
    fun saveRegion(@RequestBody dto: RegionDTO): Mono<RegionDTO> {
        return singleExecutorHelper.runOnExecutor(service.saveObjectAndPublishChanges(dto))
    }

    @PostMapping("/delete")
    fun deleteRegion(@RequestBody dto: RegionDTO): Mono<Unit> {
        return singleExecutorHelper.runOnExecutor(service.deleteObjectAndPublishChanges(dto))
    }
}